### ISSUE TITLE
3.0.0: test suite is failing in `example-12bit-shared-decompress-cmp` unit

### ISSUE THREAD

**Issue by kloczek** [libjpeg-turbo/libjpeg-turbo#705]
Looks like something is wrong with just released 3.0.0
<details>

```console
+ /usr/bin/ctest --test-dir x86_64-redhat-linux-gnu --output-on-failure --force-new-ctest-process -j48
Internal ctest changing into directory: /home/tkloczko/rpmbuild/BUILD/libjpeg-turbo-3.0.0/x86_64-redhat-linux-gnu
Test project /home/tkloczko/rpmbuild/BUILD/libjpeg-turbo-3.0.0/x86_64-redhat-linux-gnu
        Start  17: tjbench-shared-tile-cp
        Start  39: tjbench-shared-tilem-cp
        Start  51: tjbench12-shared-tile-cp
        Start  73: tjbench12-shared-tilem-cp
        Start  85: cjpeg16-shared-lossless
        Start  89: cjpeg-shared-rgb-islow
        Start 100: cjpeg-shared-422-ifast-opt
        Start 104: cjpeg-shared-440-islow
        Start 114: cjpeg-shared-420-q100-ifast-prog
        Start 120: cjpeg-shared-gray-islow
        Start 132: cjpeg-shared-3x2-float-prog
        Start 136: cjpeg-shared-3x2-ifast-prog
        Start 184: cjpeg-shared-lossless
        Start 192: cjpeg-shared-420-islow-prog
        Start 197: cjpeg-shared-444-islow
        Start 200: cjpeg-shared-444-islow-prog
        Start 203: cjpeg-shared-444-islow-ari
        Start 212: cjpeg12-shared-rgb-islow
        Start 219: cjpeg12-shared-422-ifast-opt
        Start 223: cjpeg12-shared-440-islow
        Start 229: cjpeg12-shared-420-q100-ifast-prog
        Start 235: cjpeg12-shared-gray-islow
        Start 243: cjpeg12-shared-3x2-float-prog
        Start 247: cjpeg12-shared-3x2-ifast-prog
        Start 275: cjpeg12-shared-lossless
        Start 281: cjpeg12-shared-420-islow-prog
        Start 284: cjpeg12-shared-444-islow
        Start 287: cjpeg12-shared-444-islow-prog
        Start 130: cjpeg-shared-420s-ifast-opt
        Start 140: cjpeg-shared-420-islow-ari
        Start 142: jpegtran-shared-420-islow-ari
        Start 144: cjpeg-shared-444-islow-progari
        Start 146: djpeg-shared-420m-ifast-ari
        Start 148: jpegtran-shared-420-islow
        Start 150: djpeg-shared-420m-islow-2_1
        Start 152: djpeg-shared-420m-islow-15_8
        Start 154: djpeg-shared-420m-islow-13_8
        Start 156: djpeg-shared-420m-islow-11_8
        Start 158: djpeg-shared-420m-islow-9_8
        Start 160: djpeg-shared-420m-islow-7_8
        Start 162: djpeg-shared-420m-islow-3_4
        Start 164: djpeg-shared-420m-islow-5_8
        Start 166: djpeg-shared-420m-islow-1_2
        Start 168: djpeg-shared-420m-islow-3_8
        Start 170: djpeg-shared-420m-islow-1_4
        Start 172: djpeg-shared-420m-islow-1_8
        Start 174: djpeg-shared-420-islow-256
        Start 176: djpeg-shared-420-islow-565
  1/295 Test  #17: tjbench-shared-tile-cp ..............................   Passed    0.07 sec
        Start  18: tjbench-shared-tile
  2/295 Test  #39: tjbench-shared-tilem-cp .............................   Passed    0.07 sec
        Start  40: tjbench-shared-tilem
  3/295 Test  #51: tjbench12-shared-tile-cp ............................   Passed    0.07 sec
        Start  52: tjbench12-shared-tile
  4/295 Test  #73: tjbench12-shared-tilem-cp ...........................   Passed    0.07 sec
        Start  74: tjbench12-shared-tilem
  5/295 Test  #85: cjpeg16-shared-lossless .............................   Passed    0.07 sec
        Start  87: djpeg16-shared-lossless
  6/295 Test  #89: cjpeg-shared-rgb-islow ..............................   Passed    0.08 sec
        Start  91: djpeg-shared-rgb-islow
  7/295 Test #100: cjpeg-shared-422-ifast-opt ..........................   Passed    0.08 sec
        Start  94: jpegtran-shared-icc
  8/295 Test #104: cjpeg-shared-440-islow ..............................   Passed    0.08 sec
        Start  96: djpeg-shared-rgb-islow-565
  9/295 Test #114: cjpeg-shared-420-q100-ifast-prog ....................   Passed    0.08 sec
        Start  98: djpeg-shared-rgb-islow-565D
 10/295 Test #120: cjpeg-shared-gray-islow .............................   Passed    0.08 sec
        Start 102: djpeg-shared-422-ifast
 11/295 Test #132: cjpeg-shared-3x2-float-prog .........................   Passed    0.08 sec
        Start 106: djpeg-shared-440-islow
 12/295 Test #136: cjpeg-shared-3x2-ifast-prog .........................   Passed    0.08 sec
        Start 108: djpeg-shared-422m-ifast
 13/295 Test #184: cjpeg-shared-lossless ...............................   Passed    0.08 sec
        Start 110: djpeg-shared-422m-ifast-565
 14/295 Test #192: cjpeg-shared-420-islow-prog .........................   Passed    0.08 sec
        Start 112: djpeg-shared-422m-ifast-565D
 15/295 Test #197: cjpeg-shared-444-islow ..............................   Passed    0.08 sec
        Start 116: djpeg-shared-420-q100-ifast-prog
 16/295 Test #200: cjpeg-shared-444-islow-prog .........................   Passed    0.08 sec
        Start 118: djpeg-shared-420m-q100-ifast-prog
 17/295 Test #203: cjpeg-shared-444-islow-ari ..........................   Passed    0.08 sec
        Start 122: djpeg-shared-gray-islow
 18/295 Test #212: cjpeg12-shared-rgb-islow ............................   Passed    0.08 sec
        Start 124: djpeg-shared-gray-islow-rgb
 19/295 Test #219: cjpeg12-shared-422-ifast-opt ........................   Passed    0.08 sec
        Start 126: djpeg-shared-gray-islow-565
 20/295 Test #223: cjpeg12-shared-440-islow ............................   Passed    0.08 sec
        Start 128: djpeg-shared-gray-islow-565D
 21/295 Test #229: cjpeg12-shared-420-q100-ifast-prog ..................   Passed    0.08 sec
        Start 134: djpeg-shared-3x2-float-prog
 22/295 Test #235: cjpeg12-shared-gray-islow ...........................   Passed    0.08 sec
        Start 138: djpeg-shared-3x2-ifast-prog
 23/295 Test #243: cjpeg12-shared-3x2-float-prog .......................   Passed    0.08 sec
        Start 178: djpeg-shared-420-islow-565D
 24/295 Test #247: cjpeg12-shared-3x2-ifast-prog .......................   Passed    0.08 sec
        Start 180: djpeg-shared-420m-islow-565
 25/295 Test #275: cjpeg12-shared-lossless .............................   Passed    0.08 sec
        Start 182: djpeg-shared-420m-islow-565D
 26/295 Test #281: cjpeg12-shared-420-islow-prog .......................   Passed    0.08 sec
        Start 186: djpeg-shared-lossless
 27/295 Test #284: cjpeg12-shared-444-islow ............................   Passed    0.08 sec
        Start 188: djpeg-shared-420-islow-skip15_31
 28/295 Test #287: cjpeg12-shared-444-islow-prog .......................   Passed    0.09 sec
        Start 190: djpeg-shared-420-islow-ari-skip16_139
 29/295 Test #130: cjpeg-shared-420s-ifast-opt .........................   Passed    0.09 sec
        Start 193: djpeg-shared-420-islow-prog-crop62x62_71_71
 30/295 Test #140: cjpeg-shared-420-islow-ari ..........................   Passed    0.09 sec
        Start 195: djpeg-shared-420-islow-ari-crop53x53_4_4
 31/295 Test #142: jpegtran-shared-420-islow-ari .......................   Passed    0.09 sec
        Start 198: djpeg-shared-444-islow-skip1_6
 32/295 Test #144: cjpeg-shared-444-islow-progari ......................   Passed    0.09 sec
        Start 201: djpeg-shared-444-islow-prog-crop98x98_13_13
 33/295 Test #146: djpeg-shared-420m-ifast-ari .........................   Passed    0.09 sec
        Start 204: djpeg-shared-444-islow-ari-crop37x37_0_0
 34/295 Test #148: jpegtran-shared-420-islow ...........................   Passed    0.09 sec
        Start 206: jpegtran-shared-crop
 35/295 Test #150: djpeg-shared-420m-islow-2_1 .........................   Passed    0.09 sec
        Start 208: example-8bit-shared-compress
 36/295 Test #152: djpeg-shared-420m-islow-15_8 ........................   Passed    0.09 sec
        Start 210: example-8bit-shared-decompress
 37/295 Test #154: djpeg-shared-420m-islow-13_8 ........................   Passed    0.09 sec
        Start 214: djpeg12-shared-rgb-islow
 38/295 Test #156: djpeg-shared-420m-islow-11_8 ........................   Passed    0.09 sec
        Start 217: jpegtran12-shared-icc
 39/295 Test #158: djpeg-shared-420m-islow-9_8 .........................   Passed    0.09 sec
        Start 221: djpeg12-shared-422-ifast
 40/295 Test #160: djpeg-shared-420m-islow-7_8 .........................   Passed    0.09 sec
        Start 225: djpeg12-shared-440-islow
 41/295 Test #162: djpeg-shared-420m-islow-3_4 .........................   Passed    0.09 sec
        Start 227: djpeg12-shared-422m-ifast
 42/295 Test #164: djpeg-shared-420m-islow-5_8 .........................   Passed    0.09 sec
        Start 231: djpeg12-shared-420-q100-ifast-prog
 43/295 Test #166: djpeg-shared-420m-islow-1_2 .........................   Passed    0.09 sec
        Start 233: djpeg12-shared-420m-q100-ifast-prog
 44/295 Test #168: djpeg-shared-420m-islow-3_8 .........................   Passed    0.10 sec
        Start 237: djpeg12-shared-gray-islow
 45/295 Test #170: djpeg-shared-420m-islow-1_4 .........................   Passed    0.10 sec
        Start 239: djpeg12-shared-gray-islow-rgb
 46/295 Test #172: djpeg-shared-420m-islow-1_8 .........................   Passed    0.10 sec
        Start 241: cjpeg12-shared-420s-ifast-opt
 47/295 Test #176: djpeg-shared-420-islow-565 ..........................   Passed    0.10 sec
        Start 245: djpeg12-shared-3x2-float-prog
 48/295 Test #174: djpeg-shared-420-islow-256 ..........................   Passed    0.10 sec
        Start 249: djpeg12-shared-3x2-ifast-prog
 49/295 Test  #87: djpeg16-shared-lossless .............................   Passed    0.09 sec
        Start 251: djpeg12-shared-420m-islow-2_1
 50/295 Test  #91: djpeg-shared-rgb-islow ..............................   Passed    0.09 sec
        Start 253: djpeg12-shared-420m-islow-15_8
 51/295 Test  #94: jpegtran-shared-icc .................................   Passed    0.09 sec
        Start 255: djpeg12-shared-420m-islow-13_8
 52/295 Test  #96: djpeg-shared-rgb-islow-565 ..........................   Passed    0.09 sec
        Start 257: djpeg12-shared-420m-islow-11_8
 53/295 Test  #98: djpeg-shared-rgb-islow-565D .........................   Passed    0.09 sec
        Start 259: djpeg12-shared-420m-islow-9_8
 54/295 Test #102: djpeg-shared-422-ifast ..............................   Passed    0.09 sec
        Start 261: djpeg12-shared-420m-islow-7_8
 55/295 Test #106: djpeg-shared-440-islow ..............................   Passed    0.09 sec
        Start 263: djpeg12-shared-420m-islow-3_4
 56/295 Test #108: djpeg-shared-422m-ifast .............................   Passed    0.09 sec
        Start 265: djpeg12-shared-420m-islow-5_8
 57/295 Test #110: djpeg-shared-422m-ifast-565 .........................   Passed    0.09 sec
        Start 267: djpeg12-shared-420m-islow-1_2
 58/295 Test #112: djpeg-shared-422m-ifast-565D ........................   Passed    0.09 sec
        Start 269: djpeg12-shared-420m-islow-3_8
 59/295 Test #116: djpeg-shared-420-q100-ifast-prog ....................   Passed    0.09 sec
        Start 271: djpeg12-shared-420m-islow-1_4
 60/295 Test #118: djpeg-shared-420m-q100-ifast-prog ...................   Passed    0.09 sec
        Start 273: djpeg12-shared-420m-islow-1_8
 61/295 Test #122: djpeg-shared-gray-islow .............................   Passed    0.09 sec
        Start 277: djpeg12-shared-lossless
 62/295 Test #124: djpeg-shared-gray-islow-rgb .........................   Passed    0.09 sec
        Start 279: djpeg12-shared-420-islow-skip15_31
 63/295 Test #126: djpeg-shared-gray-islow-565 .........................   Passed    0.09 sec
        Start 282: djpeg12-shared-420-islow-prog-crop62x62_71_71
 64/295 Test #128: djpeg-shared-gray-islow-565D ........................   Passed    0.09 sec
        Start 285: djpeg12-shared-444-islow-skip1_6
 65/295 Test #134: djpeg-shared-3x2-float-prog .........................   Passed    0.09 sec
        Start 288: djpeg12-shared-444-islow-prog-crop98x98_13_13
 66/295 Test #138: djpeg-shared-3x2-ifast-prog .........................   Passed    0.09 sec
        Start 290: jpegtran12-shared-crop
 67/295 Test #178: djpeg-shared-420-islow-565D .........................   Passed    0.09 sec
        Start 292: example-12bit-shared-compress
 68/295 Test #180: djpeg-shared-420m-islow-565 .........................   Passed    0.09 sec
        Start 294: example-12bit-shared-decompress
 69/295 Test #182: djpeg-shared-420m-islow-565D ........................   Passed    0.09 sec
        Start   1: tjunittest-shared
 70/295 Test #186: djpeg-shared-lossless ...............................   Passed    0.09 sec
        Start   2: tjunittest-shared-alloc
 71/295 Test #188: djpeg-shared-420-islow-skip15_31 ....................   Passed    0.09 sec
        Start   3: tjunittest-shared-yuv
 72/295 Test #190: djpeg-shared-420-islow-ari-skip16_139 ...............   Passed    0.09 sec
        Start   4: tjunittest-shared-yuv-alloc
 73/295 Test #193: djpeg-shared-420-islow-prog-crop62x62_71_71 .........   Passed    0.09 sec
        Start   5: tjunittest-shared-yuv-nopad
 74/295 Test #195: djpeg-shared-420-islow-ari-crop53x53_4_4 ............   Passed    0.09 sec
        Start   6: tjunittest-shared-lossless
 75/295 Test #198: djpeg-shared-444-islow-skip1_6 ......................   Passed    0.09 sec
        Start   7: tjunittest-shared-lossless-alloc
 76/295 Test #201: djpeg-shared-444-islow-prog-crop98x98_13_13 .........   Passed    0.09 sec
        Start   8: tjunittest-shared-bmp
 77/295 Test #204: djpeg-shared-444-islow-ari-crop37x37_0_0 ............   Passed    0.09 sec
        Start   9: tjunittest12-shared
 78/295 Test #206: jpegtran-shared-crop ................................   Passed    0.09 sec
        Start  10: tjunittest12-shared-alloc
 79/295 Test #208: example-8bit-shared-compress ........................   Passed    0.09 sec
        Start  11: tjunittest12-shared-lossless
 80/295 Test #210: example-8bit-shared-decompress ......................   Passed    0.09 sec
        Start  12: tjunittest12-shared-lossless-alloc
 81/295 Test #214: djpeg12-shared-rgb-islow ............................   Passed    0.09 sec
        Start  13: tjunittest12-shared-bmp
 82/295 Test #217: jpegtran12-shared-icc ...............................   Passed    0.09 sec
        Start  14: tjunittest16-shared-lossless
 83/295 Test #221: djpeg12-shared-422-ifast ............................   Passed    0.09 sec
        Start  15: tjunittest16-shared-lossless-alloc
 84/295 Test #225: djpeg12-shared-440-islow ............................   Passed    0.09 sec
        Start  16: tjunittest16-shared-bmp
 85/295 Test #227: djpeg12-shared-422m-ifast ...........................   Passed    0.09 sec
        Start  86: cjpeg16-shared-lossless-cmp
 86/295 Test #231: djpeg12-shared-420-q100-ifast-prog ..................   Passed    0.09 sec
        Start  88: djpeg16-shared-lossless-cmp
 87/295 Test #233: djpeg12-shared-420m-q100-ifast-prog .................   Passed    0.09 sec
        Start  90: cjpeg-shared-rgb-islow-cmp
 88/295 Test #237: djpeg12-shared-gray-islow ...........................   Passed    0.09 sec
        Start  92: djpeg-shared-rgb-islow-cmp
 89/295 Test #239: djpeg12-shared-gray-islow-rgb .......................   Passed    0.09 sec
        Start  93: djpeg-shared-rgb-islow-icc-cmp
 90/295 Test #241: cjpeg12-shared-420s-ifast-opt .......................   Passed    0.10 sec
        Start  95: jpegtran-shared-icc-cmp
 91/295 Test #245: djpeg12-shared-3x2-float-prog .......................   Passed    0.10 sec
        Start  97: djpeg-shared-rgb-islow-565-cmp
 92/295 Test #249: djpeg12-shared-3x2-ifast-prog .......................   Passed    0.10 sec
        Start  99: djpeg-shared-rgb-islow-565D-cmp
 93/295 Test #251: djpeg12-shared-420m-islow-2_1 .......................   Passed    0.10 sec
        Start 101: cjpeg-shared-422-ifast-opt-cmp
 94/295 Test #253: djpeg12-shared-420m-islow-15_8 ......................   Passed    0.10 sec
        Start 103: djpeg-shared-422-ifast-cmp
 95/295 Test #255: djpeg12-shared-420m-islow-13_8 ......................   Passed    0.10 sec
        Start 105: cjpeg-shared-440-islow-cmp
 96/295 Test #257: djpeg12-shared-420m-islow-11_8 ......................   Passed    0.10 sec
        Start 107: djpeg-shared-440-islow-cmp
 97/295 Test #259: djpeg12-shared-420m-islow-9_8 .......................   Passed    0.10 sec
        Start 109: djpeg-shared-422m-ifast-cmp
 98/295 Test #261: djpeg12-shared-420m-islow-7_8 .......................   Passed    0.10 sec
        Start 111: djpeg-shared-422m-ifast-565-cmp
 99/295 Test #263: djpeg12-shared-420m-islow-3_4 .......................   Passed    0.10 sec
        Start 113: djpeg-shared-422m-ifast-565D-cmp
100/295 Test #265: djpeg12-shared-420m-islow-5_8 .......................   Passed    0.10 sec
        Start 115: cjpeg-shared-420-q100-ifast-prog-cmp
101/295 Test #267: djpeg12-shared-420m-islow-1_2 .......................   Passed    0.10 sec
        Start 117: djpeg-shared-420-q100-ifast-prog-cmp
102/295 Test #269: djpeg12-shared-420m-islow-3_8 .......................   Passed    0.10 sec
        Start 119: djpeg-shared-420m-q100-ifast-prog-cmp
103/295 Test #271: djpeg12-shared-420m-islow-1_4 .......................   Passed    0.10 sec
        Start 121: cjpeg-shared-gray-islow-cmp
104/295 Test #273: djpeg12-shared-420m-islow-1_8 .......................   Passed    0.10 sec
        Start 123: djpeg-shared-gray-islow-cmp
105/295 Test #277: djpeg12-shared-lossless .............................   Passed    0.10 sec
        Start 125: djpeg-shared-gray-islow-rgb-cmp
106/295 Test #279: djpeg12-shared-420-islow-skip15_31 ..................   Passed    0.10 sec
        Start 127: djpeg-shared-gray-islow-565-cmp
107/295 Test #282: djpeg12-shared-420-islow-prog-crop62x62_71_71 .......   Passed    0.10 sec
        Start 129: djpeg-shared-gray-islow-565D-cmp
108/295 Test #285: djpeg12-shared-444-islow-skip1_6 ....................   Passed    0.10 sec
        Start 131: cjpeg-shared-420s-ifast-opt-cmp
109/295 Test #288: djpeg12-shared-444-islow-prog-crop98x98_13_13 .......   Passed    0.10 sec
        Start 133: cjpeg-shared-3x2-float-prog-cmp
110/295 Test #290: jpegtran12-shared-crop ..............................   Passed    0.10 sec
        Start 135: djpeg-shared-3x2-float-prog-cmp
111/295 Test #292: example-12bit-shared-compress .......................   Passed    0.10 sec
        Start 137: cjpeg-shared-3x2-ifast-prog-cmp
112/295 Test #294: example-12bit-shared-decompress .....................   Passed    0.10 sec
        Start 139: djpeg-shared-3x2-ifast-prog-cmp
113/295 Test  #86: cjpeg16-shared-lossless-cmp .........................   Passed    0.06 sec
        Start 141: cjpeg-shared-420-islow-ari-cmp
114/295 Test  #88: djpeg16-shared-lossless-cmp .........................   Passed    0.06 sec
        Start 143: jpegtran-shared-420-islow-ari-cmp
115/295 Test  #90: cjpeg-shared-rgb-islow-cmp ..........................   Passed    0.06 sec
        Start 145: cjpeg-shared-444-islow-progari-cmp
116/295 Test  #92: djpeg-shared-rgb-islow-cmp ..........................   Passed    0.06 sec
        Start 147: djpeg-shared-420m-ifast-ari-cmp
117/295 Test   #8: tjunittest-shared-bmp ...............................   Passed    0.10 sec
        Start 149: jpegtran-shared-420-islow-cmp
118/295 Test  #93: djpeg-shared-rgb-islow-icc-cmp ......................   Passed    0.07 sec
        Start 151: djpeg-shared-420m-islow-2_1-cmp
119/295 Test  #13: tjunittest12-shared-bmp .............................   Passed    0.09 sec
        Start 153: djpeg-shared-420m-islow-15_8-cmp
120/295 Test  #95: jpegtran-shared-icc-cmp .............................   Passed    0.06 sec
        Start 155: djpeg-shared-420m-islow-13_8-cmp
121/295 Test  #97: djpeg-shared-rgb-islow-565-cmp ......................   Passed    0.06 sec
        Start 157: djpeg-shared-420m-islow-11_8-cmp
122/295 Test  #99: djpeg-shared-rgb-islow-565D-cmp .....................   Passed    0.06 sec
        Start 159: djpeg-shared-420m-islow-9_8-cmp
123/295 Test #101: cjpeg-shared-422-ifast-opt-cmp ......................   Passed    0.06 sec
        Start 161: djpeg-shared-420m-islow-7_8-cmp
124/295 Test #103: djpeg-shared-422-ifast-cmp ..........................   Passed    0.06 sec
        Start 163: djpeg-shared-420m-islow-3_4-cmp
125/295 Test #105: cjpeg-shared-440-islow-cmp ..........................   Passed    0.06 sec
        Start 165: djpeg-shared-420m-islow-5_8-cmp
126/295 Test #107: djpeg-shared-440-islow-cmp ..........................   Passed    0.06 sec
        Start 167: djpeg-shared-420m-islow-1_2-cmp
127/295 Test #109: djpeg-shared-422m-ifast-cmp .........................   Passed    0.06 sec
        Start 169: djpeg-shared-420m-islow-3_8-cmp
128/295 Test #111: djpeg-shared-422m-ifast-565-cmp .....................   Passed    0.06 sec
        Start 171: djpeg-shared-420m-islow-1_4-cmp
129/295 Test #113: djpeg-shared-422m-ifast-565D-cmp ....................   Passed    0.07 sec
        Start 173: djpeg-shared-420m-islow-1_8-cmp
130/295 Test #115: cjpeg-shared-420-q100-ifast-prog-cmp ................   Passed    0.07 sec
        Start 175: djpeg-shared-420-islow-256-cmp
131/295 Test #117: djpeg-shared-420-q100-ifast-prog-cmp ................   Passed    0.07 sec
        Start 177: djpeg-shared-420-islow-565-cmp
132/295 Test #119: djpeg-shared-420m-q100-ifast-prog-cmp ...............   Passed    0.07 sec
        Start 179: djpeg-shared-420-islow-565D-cmp
133/295 Test #121: cjpeg-shared-gray-islow-cmp .........................   Passed    0.07 sec
        Start 181: djpeg-shared-420m-islow-565-cmp
134/295 Test #123: djpeg-shared-gray-islow-cmp .........................   Passed    0.07 sec
        Start 183: djpeg-shared-420m-islow-565D-cmp
135/295 Test #125: djpeg-shared-gray-islow-rgb-cmp .....................   Passed    0.07 sec
        Start 185: cjpeg-shared-lossless-cmp
136/295 Test #127: djpeg-shared-gray-islow-565-cmp .....................   Passed    0.07 sec
        Start 187: djpeg-shared-lossless-cmp
137/295 Test #129: djpeg-shared-gray-islow-565D-cmp ....................   Passed    0.07 sec
        Start 189: djpeg-shared-420-islow-skip15_31-cmp
138/295 Test #131: cjpeg-shared-420s-ifast-opt-cmp .....................   Passed    0.07 sec
        Start 191: djpeg-shared-420-islow-ari-skip16_139-cmp
139/295 Test #133: cjpeg-shared-3x2-float-prog-cmp .....................   Passed    0.07 sec
        Start 194: djpeg-shared-420-islow-prog-crop62x62_71_71-cmp
140/295 Test #135: djpeg-shared-3x2-float-prog-cmp .....................   Passed    0.07 sec
        Start 196: djpeg-shared-420-islow-ari-crop53x53_4_4-cmp
141/295 Test #137: cjpeg-shared-3x2-ifast-prog-cmp .....................   Passed    0.07 sec
        Start 199: djpeg-shared-444-islow-skip1_6-cmp
142/295 Test #139: djpeg-shared-3x2-ifast-prog-cmp .....................   Passed    0.07 sec
        Start 202: djpeg-shared-444-islow-prog-crop98x98_13_13-cmp
143/295 Test #141: cjpeg-shared-420-islow-ari-cmp ......................   Passed    0.07 sec
        Start 205: djpeg-shared-444-islow-ari-crop37x37_0_0-cmp
144/295 Test #143: jpegtran-shared-420-islow-ari-cmp ...................   Passed    0.07 sec
        Start 207: jpegtran-shared-crop-cmp
145/295 Test #145: cjpeg-shared-444-islow-progari-cmp ..................   Passed    0.07 sec
        Start 209: example-8bit-shared-compress-cmp
146/295 Test #147: djpeg-shared-420m-ifast-ari-cmp .....................   Passed    0.07 sec
        Start 211: example-8bit-shared-decompress-cmp
147/295 Test #149: jpegtran-shared-420-islow-cmp .......................   Passed    0.06 sec
        Start 213: cjpeg12-shared-rgb-islow-cmp
148/295 Test #151: djpeg-shared-420m-islow-2_1-cmp .....................   Passed    0.06 sec
        Start 215: djpeg12-shared-rgb-islow-cmp
149/295 Test #153: djpeg-shared-420m-islow-15_8-cmp ....................   Passed    0.06 sec
        Start 216: djpeg12-shared-rgb-islow-icc-cmp
150/295 Test #155: djpeg-shared-420m-islow-13_8-cmp ....................   Passed    0.06 sec
        Start 218: jpegtran12-shared-icc-cmp
151/295 Test #157: djpeg-shared-420m-islow-11_8-cmp ....................   Passed    0.06 sec
        Start 220: cjpeg12-shared-422-ifast-opt-cmp
152/295 Test #159: djpeg-shared-420m-islow-9_8-cmp .....................   Passed    0.06 sec
        Start 222: djpeg12-shared-422-ifast-cmp
153/295 Test #161: djpeg-shared-420m-islow-7_8-cmp .....................   Passed    0.06 sec
        Start 224: cjpeg12-shared-440-islow-cmp
154/295 Test #163: djpeg-shared-420m-islow-3_4-cmp .....................   Passed    0.06 sec
        Start 226: djpeg12-shared-440-islow-cmp
155/295 Test #165: djpeg-shared-420m-islow-5_8-cmp .....................   Passed    0.06 sec
        Start 228: djpeg12-shared-422m-ifast-cmp
156/295 Test #167: djpeg-shared-420m-islow-1_2-cmp .....................   Passed    0.06 sec
        Start 230: cjpeg12-shared-420-q100-ifast-prog-cmp
157/295 Test #169: djpeg-shared-420m-islow-3_8-cmp .....................   Passed    0.06 sec
        Start 232: djpeg12-shared-420-q100-ifast-prog-cmp
158/295 Test #171: djpeg-shared-420m-islow-1_4-cmp .....................   Passed    0.06 sec
        Start 234: djpeg12-shared-420m-q100-ifast-prog-cmp
159/295 Test #173: djpeg-shared-420m-islow-1_8-cmp .....................   Passed    0.06 sec
        Start 236: cjpeg12-shared-gray-islow-cmp
160/295 Test #175: djpeg-shared-420-islow-256-cmp ......................   Passed    0.06 sec
        Start 238: djpeg12-shared-gray-islow-cmp
161/295 Test #177: djpeg-shared-420-islow-565-cmp ......................   Passed    0.06 sec
        Start 240: djpeg12-shared-gray-islow-rgb-cmp
162/295 Test #179: djpeg-shared-420-islow-565D-cmp .....................   Passed    0.06 sec
        Start 242: cjpeg12-shared-420s-ifast-opt-cmp
163/295 Test #181: djpeg-shared-420m-islow-565-cmp .....................   Passed    0.06 sec
        Start 244: cjpeg12-shared-3x2-float-prog-cmp
164/295 Test #183: djpeg-shared-420m-islow-565D-cmp ....................   Passed    0.06 sec
        Start 246: djpeg12-shared-3x2-float-prog-cmp
165/295 Test #185: cjpeg-shared-lossless-cmp ...........................   Passed    0.06 sec
        Start 248: cjpeg12-shared-3x2-ifast-prog-cmp
166/295 Test #187: djpeg-shared-lossless-cmp ...........................   Passed    0.06 sec
        Start 250: djpeg12-shared-3x2-ifast-prog-cmp
167/295 Test #189: djpeg-shared-420-islow-skip15_31-cmp ................   Passed    0.06 sec
        Start 252: djpeg12-shared-420m-islow-2_1-cmp
168/295 Test #191: djpeg-shared-420-islow-ari-skip16_139-cmp ...........   Passed    0.06 sec
        Start 254: djpeg12-shared-420m-islow-15_8-cmp
169/295 Test #194: djpeg-shared-420-islow-prog-crop62x62_71_71-cmp .....   Passed    0.06 sec
        Start 256: djpeg12-shared-420m-islow-13_8-cmp
170/295 Test #196: djpeg-shared-420-islow-ari-crop53x53_4_4-cmp ........   Passed    0.06 sec
        Start 258: djpeg12-shared-420m-islow-11_8-cmp
171/295 Test #199: djpeg-shared-444-islow-skip1_6-cmp ..................   Passed    0.06 sec
        Start 260: djpeg12-shared-420m-islow-9_8-cmp
172/295 Test #202: djpeg-shared-444-islow-prog-crop98x98_13_13-cmp .....   Passed    0.06 sec
        Start 262: djpeg12-shared-420m-islow-7_8-cmp
173/295 Test #205: djpeg-shared-444-islow-ari-crop37x37_0_0-cmp ........   Passed    0.06 sec
        Start 264: djpeg12-shared-420m-islow-3_4-cmp
174/295 Test #207: jpegtran-shared-crop-cmp ............................   Passed    0.06 sec
        Start 266: djpeg12-shared-420m-islow-5_8-cmp
175/295 Test #209: example-8bit-shared-compress-cmp ....................   Passed    0.06 sec
        Start 268: djpeg12-shared-420m-islow-1_2-cmp
176/295 Test  #16: tjunittest16-shared-bmp .............................   Passed    0.20 sec
        Start 270: djpeg12-shared-420m-islow-3_8-cmp
177/295 Test #211: example-8bit-shared-decompress-cmp ..................   Passed    0.06 sec
        Start 272: djpeg12-shared-420m-islow-1_4-cmp
178/295 Test #213: cjpeg12-shared-rgb-islow-cmp ........................   Passed    0.06 sec
        Start 274: djpeg12-shared-420m-islow-1_8-cmp
179/295 Test #215: djpeg12-shared-rgb-islow-cmp ........................   Passed    0.06 sec
        Start 276: cjpeg12-shared-lossless-cmp
180/295 Test #216: djpeg12-shared-rgb-islow-icc-cmp ....................   Passed    0.06 sec
        Start 278: djpeg12-shared-lossless-cmp
181/295 Test #218: jpegtran12-shared-icc-cmp ...........................   Passed    0.06 sec
        Start 280: djpeg12-shared-420-islow-skip15_31-cmp
182/295 Test #220: cjpeg12-shared-422-ifast-opt-cmp ....................   Passed    0.06 sec
        Start 283: djpeg12-shared-420-islow-prog-crop62x62_71_71-cmp
183/295 Test #222: djpeg12-shared-422-ifast-cmp ........................   Passed    0.06 sec
        Start 286: djpeg12-shared-444-islow-skip1_6-cmp
184/295 Test #224: cjpeg12-shared-440-islow-cmp ........................   Passed    0.06 sec
        Start 289: djpeg12-shared-444-islow-prog-crop98x98_13_13-cmp
185/295 Test #226: djpeg12-shared-440-islow-cmp ........................   Passed    0.06 sec
        Start 291: jpegtran12-shared-crop-cmp
186/295 Test #228: djpeg12-shared-422m-ifast-cmp .......................   Passed    0.06 sec
        Start 293: example-12bit-shared-compress-cmp
187/295 Test #230: cjpeg12-shared-420-q100-ifast-prog-cmp ..............   Passed    0.06 sec
        Start 295: example-12bit-shared-decompress-cmp
188/295 Test #232: djpeg12-shared-420-q100-ifast-prog-cmp ..............   Passed    0.06 sec
189/295 Test #234: djpeg12-shared-420m-q100-ifast-prog-cmp .............   Passed    0.06 sec
190/295 Test #236: cjpeg12-shared-gray-islow-cmp .......................   Passed    0.06 sec
191/295 Test #238: djpeg12-shared-gray-islow-cmp .......................   Passed    0.06 sec
192/295 Test #240: djpeg12-shared-gray-islow-rgb-cmp ...................   Passed    0.05 sec
193/295 Test #242: cjpeg12-shared-420s-ifast-opt-cmp ...................   Passed    0.05 sec
194/295 Test #244: cjpeg12-shared-3x2-float-prog-cmp ...................   Passed    0.05 sec
195/295 Test #246: djpeg12-shared-3x2-float-prog-cmp ...................   Passed    0.05 sec
196/295 Test #248: cjpeg12-shared-3x2-ifast-prog-cmp ...................   Passed    0.05 sec
197/295 Test #250: djpeg12-shared-3x2-ifast-prog-cmp ...................   Passed    0.05 sec
198/295 Test #252: djpeg12-shared-420m-islow-2_1-cmp ...................   Passed    0.04 sec
199/295 Test #254: djpeg12-shared-420m-islow-15_8-cmp ..................   Passed    0.04 sec
200/295 Test #256: djpeg12-shared-420m-islow-13_8-cmp ..................   Passed    0.04 sec
201/295 Test #258: djpeg12-shared-420m-islow-11_8-cmp ..................   Passed    0.04 sec
202/295 Test #260: djpeg12-shared-420m-islow-9_8-cmp ...................   Passed    0.04 sec
203/295 Test #262: djpeg12-shared-420m-islow-7_8-cmp ...................   Passed    0.04 sec
204/295 Test #264: djpeg12-shared-420m-islow-3_4-cmp ...................   Passed    0.03 sec
205/295 Test #266: djpeg12-shared-420m-islow-5_8-cmp ...................   Passed    0.03 sec
206/295 Test #268: djpeg12-shared-420m-islow-1_2-cmp ...................   Passed    0.03 sec
207/295 Test #270: djpeg12-shared-420m-islow-3_8-cmp ...................   Passed    0.03 sec
208/295 Test #272: djpeg12-shared-420m-islow-1_4-cmp ...................   Passed    0.03 sec
209/295 Test #274: djpeg12-shared-420m-islow-1_8-cmp ...................   Passed    0.02 sec
210/295 Test #276: cjpeg12-shared-lossless-cmp .........................   Passed    0.02 sec
211/295 Test #278: djpeg12-shared-lossless-cmp .........................   Passed    0.02 sec
212/295 Test #280: djpeg12-shared-420-islow-skip15_31-cmp ..............   Passed    0.02 sec
213/295 Test #283: djpeg12-shared-420-islow-prog-crop62x62_71_71-cmp ...   Passed    0.02 sec
214/295 Test #286: djpeg12-shared-444-islow-skip1_6-cmp ................   Passed    0.02 sec
215/295 Test #289: djpeg12-shared-444-islow-prog-crop98x98_13_13-cmp ...   Passed    0.01 sec
216/295 Test #291: jpegtran12-shared-crop-cmp ..........................   Passed    0.01 sec
217/295 Test #293: example-12bit-shared-compress-cmp ...................   Passed    0.01 sec
218/295 Test #295: example-12bit-shared-decompress-cmp .................***Failed    0.01 sec
testout12-example.ppm: FAILED.  Checksum is d41d8cd98f00b204e9800998ecf8427e

219/295 Test  #18: tjbench-shared-tile .................................   Passed    0.57 sec
        Start  19: tjbench-shared-tile-gray-8x8-cmp
        Start  20: tjbench-shared-tile-420-8x8-cmp
        Start  21: tjbench-shared-tile-422-8x8-cmp
        Start  22: tjbench-shared-tile-444-8x8-cmp
        Start  23: tjbench-shared-tile-gray-16x16-cmp
        Start  24: tjbench-shared-tile-420-16x16-cmp
        Start  25: tjbench-shared-tile-422-16x16-cmp
        Start  26: tjbench-shared-tile-444-16x16-cmp
        Start  27: tjbench-shared-tile-gray-32x32-cmp
        Start  28: tjbench-shared-tile-420-32x32-cmp
        Start  29: tjbench-shared-tile-422-32x32-cmp
        Start  30: tjbench-shared-tile-444-32x32-cmp
        Start  31: tjbench-shared-tile-gray-64x64-cmp
        Start  32: tjbench-shared-tile-420-64x64-cmp
        Start  33: tjbench-shared-tile-422-64x64-cmp
        Start  34: tjbench-shared-tile-444-64x64-cmp
        Start  35: tjbench-shared-tile-gray-128x128-cmp
        Start  36: tjbench-shared-tile-420-128x128-cmp
        Start  37: tjbench-shared-tile-422-128x128-cmp
        Start  38: tjbench-shared-tile-444-128x128-cmp
220/295 Test  #40: tjbench-shared-tilem ................................   Passed    0.60 sec
        Start  41: tjbench-shared-tile-420m-8x8-cmp
        Start  42: tjbench-shared-tile-422m-8x8-cmp
        Start  43: tjbench-shared-tile-420m-16x16-cmp
        Start  44: tjbench-shared-tile-422m-16x16-cmp
        Start  45: tjbench-shared-tile-420m-32x32-cmp
        Start  46: tjbench-shared-tile-422m-32x32-cmp
        Start  47: tjbench-shared-tile-420m-64x64-cmp
        Start  48: tjbench-shared-tile-422m-64x64-cmp
        Start  49: tjbench-shared-tile-420m-128x128-cmp
        Start  50: tjbench-shared-tile-422m-128x128-cmp
221/295 Test  #19: tjbench-shared-tile-gray-8x8-cmp ....................   Passed    0.05 sec
222/295 Test  #20: tjbench-shared-tile-420-8x8-cmp .....................   Passed    0.05 sec
223/295 Test  #21: tjbench-shared-tile-422-8x8-cmp .....................   Passed    0.05 sec
224/295 Test  #22: tjbench-shared-tile-444-8x8-cmp .....................   Passed    0.05 sec
225/295 Test  #23: tjbench-shared-tile-gray-16x16-cmp ..................   Passed    0.05 sec
226/295 Test  #24: tjbench-shared-tile-420-16x16-cmp ...................   Passed    0.05 sec
227/295 Test  #25: tjbench-shared-tile-422-16x16-cmp ...................   Passed    0.05 sec
228/295 Test  #26: tjbench-shared-tile-444-16x16-cmp ...................   Passed    0.04 sec
229/295 Test  #27: tjbench-shared-tile-gray-32x32-cmp ..................   Passed    0.04 sec
230/295 Test  #28: tjbench-shared-tile-420-32x32-cmp ...................   Passed    0.04 sec
231/295 Test  #29: tjbench-shared-tile-422-32x32-cmp ...................   Passed    0.04 sec
232/295 Test  #30: tjbench-shared-tile-444-32x32-cmp ...................   Passed    0.04 sec
233/295 Test  #31: tjbench-shared-tile-gray-64x64-cmp ..................   Passed    0.04 sec
234/295 Test  #32: tjbench-shared-tile-420-64x64-cmp ...................   Passed    0.03 sec
235/295 Test  #33: tjbench-shared-tile-422-64x64-cmp ...................   Passed    0.03 sec
236/295 Test  #34: tjbench-shared-tile-444-64x64-cmp ...................   Passed    0.03 sec
237/295 Test  #35: tjbench-shared-tile-gray-128x128-cmp ................   Passed    0.03 sec
238/295 Test  #36: tjbench-shared-tile-420-128x128-cmp .................   Passed    0.03 sec
239/295 Test  #37: tjbench-shared-tile-422-128x128-cmp .................   Passed    0.03 sec
240/295 Test  #38: tjbench-shared-tile-444-128x128-cmp .................   Passed    0.03 sec
241/295 Test  #41: tjbench-shared-tile-420m-8x8-cmp ....................   Passed    0.02 sec
242/295 Test  #42: tjbench-shared-tile-422m-8x8-cmp ....................   Passed    0.02 sec
243/295 Test  #43: tjbench-shared-tile-420m-16x16-cmp ..................   Passed    0.02 sec
244/295 Test  #44: tjbench-shared-tile-422m-16x16-cmp ..................   Passed    0.02 sec
245/295 Test  #45: tjbench-shared-tile-420m-32x32-cmp ..................   Passed    0.02 sec
246/295 Test  #46: tjbench-shared-tile-422m-32x32-cmp ..................   Passed    0.02 sec
247/295 Test  #47: tjbench-shared-tile-420m-64x64-cmp ..................   Passed    0.01 sec
248/295 Test  #48: tjbench-shared-tile-422m-64x64-cmp ..................   Passed    0.01 sec
249/295 Test  #49: tjbench-shared-tile-420m-128x128-cmp ................   Passed    0.01 sec
250/295 Test  #50: tjbench-shared-tile-422m-128x128-cmp ................   Passed    0.01 sec
251/295 Test  #74: tjbench12-shared-tilem ..............................   Passed    0.92 sec
        Start  75: tjbench12-shared-tile-420m-8x8-cmp
        Start  76: tjbench12-shared-tile-422m-8x8-cmp
        Start  77: tjbench12-shared-tile-420m-16x16-cmp
        Start  78: tjbench12-shared-tile-422m-16x16-cmp
        Start  79: tjbench12-shared-tile-420m-32x32-cmp
        Start  80: tjbench12-shared-tile-422m-32x32-cmp
        Start  81: tjbench12-shared-tile-420m-64x64-cmp
        Start  82: tjbench12-shared-tile-422m-64x64-cmp
        Start  83: tjbench12-shared-tile-420m-128x128-cmp
        Start  84: tjbench12-shared-tile-422m-128x128-cmp
252/295 Test  #75: tjbench12-shared-tile-420m-8x8-cmp ..................   Passed    0.02 sec
253/295 Test  #76: tjbench12-shared-tile-422m-8x8-cmp ..................   Passed    0.02 sec
254/295 Test  #77: tjbench12-shared-tile-420m-16x16-cmp ................   Passed    0.02 sec
255/295 Test  #78: tjbench12-shared-tile-422m-16x16-cmp ................   Passed    0.02 sec
256/295 Test  #79: tjbench12-shared-tile-420m-32x32-cmp ................   Passed    0.01 sec
257/295 Test  #80: tjbench12-shared-tile-422m-32x32-cmp ................   Passed    0.01 sec
258/295 Test  #81: tjbench12-shared-tile-420m-64x64-cmp ................   Passed    0.01 sec
259/295 Test  #82: tjbench12-shared-tile-422m-64x64-cmp ................   Passed    0.01 sec
260/295 Test  #83: tjbench12-shared-tile-420m-128x128-cmp ..............   Passed    0.01 sec
261/295 Test  #84: tjbench12-shared-tile-422m-128x128-cmp ..............   Passed    0.01 sec
262/295 Test  #52: tjbench12-shared-tile ...............................   Passed    0.95 sec
        Start  53: tjbench12-shared-tile-gray-8x8-cmp
        Start  54: tjbench12-shared-tile-420-8x8-cmp
        Start  55: tjbench12-shared-tile-422-8x8-cmp
        Start  56: tjbench12-shared-tile-444-8x8-cmp
        Start  57: tjbench12-shared-tile-gray-16x16-cmp
        Start  58: tjbench12-shared-tile-420-16x16-cmp
        Start  59: tjbench12-shared-tile-422-16x16-cmp
        Start  60: tjbench12-shared-tile-444-16x16-cmp
        Start  61: tjbench12-shared-tile-gray-32x32-cmp
        Start  62: tjbench12-shared-tile-420-32x32-cmp
        Start  63: tjbench12-shared-tile-422-32x32-cmp
        Start  64: tjbench12-shared-tile-444-32x32-cmp
        Start  65: tjbench12-shared-tile-gray-64x64-cmp
        Start  66: tjbench12-shared-tile-420-64x64-cmp
        Start  67: tjbench12-shared-tile-422-64x64-cmp
        Start  68: tjbench12-shared-tile-444-64x64-cmp
        Start  69: tjbench12-shared-tile-gray-128x128-cmp
        Start  70: tjbench12-shared-tile-420-128x128-cmp
        Start  71: tjbench12-shared-tile-422-128x128-cmp
        Start  72: tjbench12-shared-tile-444-128x128-cmp
263/295 Test  #53: tjbench12-shared-tile-gray-8x8-cmp ..................   Passed    0.05 sec
264/295 Test  #54: tjbench12-shared-tile-420-8x8-cmp ...................   Passed    0.05 sec
265/295 Test  #55: tjbench12-shared-tile-422-8x8-cmp ...................   Passed    0.04 sec
266/295 Test  #56: tjbench12-shared-tile-444-8x8-cmp ...................   Passed    0.04 sec
267/295 Test  #57: tjbench12-shared-tile-gray-16x16-cmp ................   Passed    0.04 sec
268/295 Test  #58: tjbench12-shared-tile-420-16x16-cmp .................   Passed    0.04 sec
269/295 Test  #59: tjbench12-shared-tile-422-16x16-cmp .................   Passed    0.04 sec
270/295 Test  #60: tjbench12-shared-tile-444-16x16-cmp .................   Passed    0.03 sec
271/295 Test  #61: tjbench12-shared-tile-gray-32x32-cmp ................   Passed    0.03 sec
272/295 Test  #62: tjbench12-shared-tile-420-32x32-cmp .................   Passed    0.03 sec
273/295 Test  #63: tjbench12-shared-tile-422-32x32-cmp .................   Passed    0.03 sec
274/295 Test  #64: tjbench12-shared-tile-444-32x32-cmp .................   Passed    0.03 sec
275/295 Test  #65: tjbench12-shared-tile-gray-64x64-cmp ................   Passed    0.02 sec
276/295 Test  #66: tjbench12-shared-tile-420-64x64-cmp .................   Passed    0.02 sec
277/295 Test  #67: tjbench12-shared-tile-422-64x64-cmp .................   Passed    0.02 sec
278/295 Test  #68: tjbench12-shared-tile-444-64x64-cmp .................   Passed    0.02 sec
279/295 Test  #69: tjbench12-shared-tile-gray-128x128-cmp ..............   Passed    0.01 sec
280/295 Test  #70: tjbench12-shared-tile-420-128x128-cmp ...............   Passed    0.01 sec
281/295 Test  #71: tjbench12-shared-tile-422-128x128-cmp ...............   Passed    0.01 sec
282/295 Test  #72: tjbench12-shared-tile-444-128x128-cmp ...............   Passed    0.01 sec
283/295 Test  #15: tjunittest16-shared-lossless-alloc ..................   Passed    2.37 sec
284/295 Test   #6: tjunittest-shared-lossless ..........................   Passed    2.60 sec
285/295 Test  #14: tjunittest16-shared-lossless ........................   Passed    2.59 sec
286/295 Test  #11: tjunittest12-shared-lossless ........................   Passed    2.69 sec
287/295 Test   #7: tjunittest-shared-lossless-alloc ....................   Passed    2.70 sec
288/295 Test  #12: tjunittest12-shared-lossless-alloc ..................   Passed    2.79 sec
289/295 Test   #4: tjunittest-shared-yuv-alloc .........................   Passed    6.73 sec
290/295 Test   #5: tjunittest-shared-yuv-nopad .........................   Passed    6.75 sec
291/295 Test   #3: tjunittest-shared-yuv ...............................   Passed    6.77 sec
292/295 Test   #1: tjunittest-shared ...................................   Passed    8.64 sec
293/295 Test   #2: tjunittest-shared-alloc .............................   Passed    8.86 sec
294/295 Test  #10: tjunittest12-shared-alloc ...........................   Passed   16.84 sec
295/295 Test   #9: tjunittest12-shared .................................   Passed   17.65 sec

99% tests passed, 1 tests failed out of 295

Total Test time (real) =  17.89 sec

The following tests FAILED:
        295 - example-12bit-shared-decompress-cmp (Failed)
Errors while running CTest
```
</details>

**Comment by chenrui333** [comment]
I am seeing two test failures as well as in CI

```
The following tests FAILED:
	246 - djpeg12-shared-3x2-float-prog-cmp (Failed)
	541 - djpeg12-static-3x2-float-prog-cmp (Failed)
```

relates to https://github.com/Homebrew/homebrew-core/pull/135676

**Comment by kloczek** [comment]
Hmm .. why those units are downloading dist tar ball? 🤔

**Comment by dcommander** [comment]
We have seen no such failures in our own CI system, which has been continuously testing the code for the entire five months that libjpeg-turbo 3.0 has been in beta.  In other to reproduce the failures, I would need to know more about your build environment.  The unit tests do not download a distribution tarball.

**Comment by diizzyy** [comment]
Also reproducable on FreeBSD 13.2-RELEASE (amd64)

It fails when (in my case) ```-march=tigerlake``` is defined but works without
LLVM/Clang version 14.0.5

**Comment by heirecka** [comment]
I see 295 - example-12bit-shared-decompress-cmp (Failed) here as well

```
The following tests FAILED:
        295 - example-12bit-shared-decompress-cmp (Failed)
Errors while running CTest

        Start 295: example-12bit-shared-decompress-cmp

295: Test command: /var/tmp/paludis/build/media-libs-libjpeg-turbo-3.0.0/work/build/md5/md5cmp "2ff0e8505ee6e0ffaeb24037d5650b57" "testout12-example.ppm"
295: Working Directory: /var/tmp/paludis/build/media-libs-libjpeg-turbo-3.0.0/work/build
295: Test timeout computed to be: 10000000
295: testout12-example.ppm: FAILED.  Checksum is d41d8cd98f00b204e9800998ecf8427e
250/295 Test #295: example-12bit-shared-decompress-cmp .................***Failed    0.01 sec
testout12-example.ppm: FAILED.  Checksum is d41d8cd98f00b204e9800998ecf8427e
```

Maybe it's a problem with running the tests in parallel? Or at least they reliably pass if I pass -j1 to ctest.

**Comment by diizzyy** [comment]
Fails with ```-j1``` on my end

**Comment by dcommander** [comment]
The floating point test failures are probably compiler-specific.  Our build system sets a default value for `FLOATTEST12` for x86-64. That default value seems to be appropriate for the compilers I've tested, but maybe it isn't appropriate for all of them.  If you can show me the complete output of the failed tests, then I can tell you the appropriate value of `FLOATTEST12` for your system.

As far as the `example-12bit-shared-decompress-cmp` failure, I have no idea.  I will repeat:

> In other to reproduce the failures, I would need to know more about your build environment.

No one has yet provided that information.

**Comment by thesamesam** [comment]
What information specifically do you want? https://github.com/libjpeg-turbo/libjpeg-turbo/issues/705#issuecomment-1619122771 seemed to give enough so I didn't bother looking myself yet.

**Comment by dcommander** [comment]
> What information specifically do you want? [#705 (comment)](https://github.com/libjpeg-turbo/libjpeg-turbo/issues/705#issuecomment-1619122771) seemed to give enough so I didn't bother looking myself yet. I can do if it'd be useful if you can't reproduce with that `-march=`.

It would be a lot more useful for the OP to answer my question about their test environment than for everyone else to pile on.  I cannot talk to five people at once.  https://github.com/libjpeg-turbo/libjpeg-turbo/issues/705#issuecomment-1619122771 describes testing with FreeBSD 13, and I don't think I have a test environment for that, so I would like to know how to repro this with Linux.

**Comment by carlocab** [comment]
> I am seeing two test failures as well as in CI
> 
> ```
> The following tests FAILED:
> 	246 - djpeg12-shared-3x2-float-prog-cmp (Failed)
> 	541 - djpeg12-static-3x2-float-prog-cmp (Failed)
> ```
> 
> relates to [Homebrew/homebrew-core#135676](https://github.com/Homebrew/homebrew-core/pull/135676)

This test failure occurs on macOS 13 on arm64 (Apple Silicon) with Xcode 14.3. We invoke `cmake` with
```
cmake -S . -B build \
  -DWITH_JPEG8=1 \
  -DCMAKE_INSTALL_PREFIX=/opt/homebrew/Cellar/jpeg-turbo/3.0.0 \
  -DCMAKE_INSTALL_LIBDIR=/opt/homebrew/Cellar/jpeg-turbo/3.0.0/lib \
  -DCMAKE_BUILD_TYPE=Release \
  -DCMAKE_FIND_FRAMEWORK=LAST \
  -DCMAKE_VERBOSE_MAKEFILE=ON \
  -Wno-dev \
  -DBUILD_TESTING=OFF \
  -DCMAKE_OSX_SYSROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk
```

We also build with the exact same configuration on Linux (Ubuntu 22.04, x86_64), macOS 13-x86_64 (also with Xcode 14.3), macOS 12 (both x86_64 and arm64, Xcode 14.2) and macOS 11 (both x86_64 and arm64, Xcode 13.2), but did not encounter any failures on these setups.

Complete build logs are available at https://github.com/Homebrew/homebrew-core/actions/runs/5447007381/jobs/9916243774?pr=135676.

This means, unfortunately, that this particular failure will likely not reproduce on Linux. Please let us know what else we can do to resolve this. How important are the failing test cases?

**Comment by dcommander** [comment]
> This means, unfortunately, that this particular failure will likely not reproduce on Linux. Please let us know what else we can do to resolve this. How important are the failing test cases?

The floating point tests are not important.  They test the floating point DCT/IDCT algorithms, which are a legacy feature.  (Back in 1991, when Tom Lane first released libjpeg, some CPUs had floating point units and some didn't.  It could take minutes to compress or decompress a 1-megapixel JPEG image using the "slow integer" DCT/IDCT algorithms, and the floating point algorithms were significantly faster on systems that had an FPU.  On systems without FPUs, floating point math was emulated and really slow, so Tom also developed "fast integer" DCT/IDCT algorithms to speed up JPEG performance, at the expense of accuracy, on those systems.  In 2023, the floating point algorithms are significantly slower than the "slow integer" algorithms without being significantly more accurate, and the "fast integer" algorithms fail the ISO/ITU-T conformance tests without being any faster than the "slow integer" algorithms on x86 systems.  Thus, the floating point and "fast integer" algorithms are considered legacy features.)  The problem with the floating point algorithms vis-a-vis regression testing is that their output can vary based on how the compiler implements floating point math (which can vary based on the compiler settings.)  The results for 8-bit data precision on x86 systems are deterministic, because there is a SIMD (SSE) implementation of the 8-bit-per-sample floating point algorithms for x86-64 and i386.  For other architectures and for 12-bit data precision, however, the build system has to make an intelligent guess as to the expected floating point results, if it can.  (It doesn't even try for i386, because there are too many possibilities.)  You can override the build system's guess by setting the `FLOATTEST8` or `FLOATTEST12` CMake variable to one of the following values:

https://github.com/libjpeg-turbo/libjpeg-turbo/blob/6c87537f60941f3c265c339fe60d1e31d2a42ccf/CMakeLists.txt#L868-L889

For arm64 builds, the build system should set `FLOATTEST12` to `fp-contract` by default, which is apparently correct in most cases but isn't correct with Xcode 14.3, for some reason.  Try passing `-DFLOATTEST12=no-fp-contract` to CMake and see if that resolves the issue.  If not, then send me the output of `ctest -V` so I can see the MD5 sums that the floating point tests are generating.

Now, as far as the `example-12bit` test failure that the OP reported, that is a different issue.  That test is also not critical, but I need to understand why it is failing on certain builds.  That test does not use floating point math, so the issue is unrelated to the floating point test failures.

**Comment by dcommander** [comment]
Also, this is what I mean when I say that I can't talk to five people at once.  It isn't always clear from the comments above who is experiencing which of the two separate issues and which of the issues were resolved by disabling parallel testing.  If the `example-12bit` failure is resolved with `-j1`, then that is just a build/test system dependency buglet that should be easy to fix.

**Comment by heirecka** [comment]
> Also, this is what I mean when I say that I can't talk to five people at once. It isn't always clear from the comments above who is experiencing which of the two separate issues and which of the issues were resolved by disabling parallel testing. If the `example-12bit` failure is resolved with `-j1`, then that is just a build/test system dependency buglet that should be easy to fix.

Yes, I posted here as it seems the same issue or at least affects the same test as the OP. Sorry, if I added confusion.
And yes the same test doesn't fail with -j1. If still needed here's the full log: https://dpaste.com/7X2Y94VQE

**Comment by diizzyy** [comment]
@dcommander

If you want a quick test setup, fire up a VM (1-2G of RAM will do if you don't use ZFS as filesystem).
* Do a basic install (takes a few minutes, boot only or memstick is fine) of 13.2 (13.1 is soon EoL), don't install the ports tree
* Run ```pkg install git``` as root
* Clone the ports tree (will take a while) as root: ```cd /usr && git clone https://github.com/freebsd/freebsd-ports.git ports```
* If you want to use precompiled dependencies: ```pkg install jpeg-turbo && pkg delete jpeg-turbo``` (as root)
* If you want to use the latest versions of dependencies available (may need more memory than 2G depending on amount of cores available as you will be building all deps from source): ```pkg delete -a -f && cd /usr/ports/graphics/jpeg-turbo && make BATCH=yes DISABLE_VULNERABILITIES=yes depends clean clean-depends```
* I've attached a patch file for 3.0.0 that you can apply to the ports tree
* Run ```make test``` in /usr/ports/graphics/jpeg-turbo to build and execute unit tests. ```make clean``` to clean up (remove work directory)
* To adjust -march switch you define ```CPUTYPE?=tigerlake``` (for example) in ```/etc/make.conf```

[jpeg-turbo-300.patch](https://github.com/libjpeg-turbo/libjpeg-turbo/files/11951611/jpeg-turbo-300.patch)

**Comment by dcommander** [comment]
The issue originally reported by @kloczek was indeed a test concurrency issue, which has been fixed (along with several other test concurrency issues.)

The floating point test failures are unrelated, and I think those failures are related to:
https://discourse.llvm.org/t/fp-contraction-fma-on-by-default/64975
With Clang 14, `-ffp-contract=on` is supposedly now the default, which means that `FLOATTEST12` should be set to `fp-contract` in order for the tests to pass.  However, in my testing on Ubuntu 22.04/x86-64, `-ffp-contract` doesn't appear to have any effect on the floating point DCT/IDCT results.  That would explain why the macOS 13/x86-64 Homebrew CI build passed while the macOS 13/arm64 Homebrew CI build failed.  If you are running into those problems, then add  `-DFLOATTEST12=fp-contract` to the CMake command line for the build in question.  (NOTE: I tried several years ago to figure out how to automatically set the `FLOATTEST*` variables, but I wasn't able to figure it out after many wasted hours.)

**Comment by chenrui333** [comment]
Looks like I can still reproduce it on homebrew CI for 13-arm

https://github.com/Homebrew/homebrew-core/actions/runs/5549879272/jobs/10160053957?pr=135676

**Comment by dcommander** [comment]
> Looks like I can still reproduce it on homebrew CI for 13-arm
> 
> 
> 
> https://github.com/Homebrew/homebrew-core/actions/runs/5549879272/jobs/10160053957?pr=135676

Please read my comments above. The Homebrew issue was not automatically fixed. You have to change the value of the CMake `FLOATTEST12` variable in the failing configuration.

**Comment by chenrui333** [comment]
noted, thanks, doing the cmake change now.

**Comment by dcommander** [comment]
@chenrui333 Then you'll need to send me the verbose output of the failing tests, which will show the expected vs. actual MD5 sum of the generated test images.  I asked for that information already (https://github.com/libjpeg-turbo/libjpeg-turbo/issues/705#issuecomment-1619193256), but because people chose to pile on to this issue with reports of an unrelated issue, my comment was probably buried in the noise.

**Comment by carlocab** [comment]
@dcommander, apologies for muddying up this issue with a different one. I'll try your suggestions from https://github.com/libjpeg-turbo/libjpeg-turbo/issues/705#issuecomment-1620306725 and open a new issue to follow it up.

**Comment by thesamesam** [comment]
> The issue originally reported by @kloczek was indeed a test concurrency issue, which has been fixed (along with several other test concurrency issues.)
> 
> The floating point test failures are unrelated, and I think those failures are related to: [discourse.llvm.org/t/fp-contraction-fma-on-by-default/64975](https://discourse.llvm.org/t/fp-contraction-fma-on-by-default/64975) With Clang 14, `-ffp-contract=on` is supposedly now the default, which means that `FLOATTEST12` should be set to `fp-contract` in order for the tests to pass. However, in my testing on Ubuntu 22.04/x86-64, `-ffp-contract` doesn't appear to have any effect on the floating point DCT/IDCT results. That would explain why the macOS 13/x86-64 Homebrew CI build passed while the macOS 13/arm64 Homebrew CI build failed. If you are running into those problems, then add `-DFLOATTEST12=fp-contract` to the CMake command line for the build in question. (NOTE: I tried several years ago to figure out how to automatically set the `FLOATTEST*` variables, but I wasn't able to figure it out after many wasted hours.)

FWIW, there's active discussion at https://gcc.gnu.org/pipermail/gcc/2023-September/242466.html as to whether GCC should change its default behaviour.

**Comment by dcommander** [comment]
I performed some experiments in an attempt to figure out appropriate default values of `FLOATTEST8` and `FLOATTEST12` for various compilers when FMA instructions are enabled.  Here are my observations when building libjpeg-turbo on Linux/x86-64 with `-march=x86-64-v3` (Clang 12+, GCC 10+) or `-march=haswell` (Clang 11-, GCC 9-) and with the SIMD extensions disabled:

Clang 14
--------

No `-ffp-contract` argument (equivalent of `-ffp-contract=on`):

The MD5 sum of the `djpeg12-*-3x2-float-prog` test is 2da9de6ae869e88b8372de815d366b03, which does not match any of the predefined `MD5_PPM_3x2_FLOAT*` constants.  However, the MD5 sums of other tests match the predefined values for `FLOATTEST8=no-fp-contract` and `FLOATTEST12=no-fp-contract`.  Referring to #709, this is consistent with the default behavior of Clang 14 when compiling AArch64 code.

`-ffp-contract=fast`:

The MD5 sum of the `djpeg12-*-3x2-float-prog` test is 03b81c43c652307ac2134885b14b43e0, which does not match any of the predefined `MD5_PPM_3x2_FLOAT*` constants.  However, the MD5 sums of other tests match the predefined values for `FLOATTEST8=no-fp-contract` and `FLOATTEST12=no-fp-contract`.

`-ffp-contract=off`:

The MD5 sums of all tests match the predefined values for `FLOATTEST8=no-fp-contract` and `FLOATTEST12=no-fp-contract`.

It seems that Clang 14 is at least consistent between AArch64 and x86-64 as long as FMA instructions are enabled on the latter.

Clang 5 through 13
------------------

Same as the behavior of Clang 14, except `-ffp-contract=off` is the default rather than `-ffp-contract=on`.

GCC 5 through 13
----------------

No `-ffp-contract` argument (equivalent of `-ffp-contract=fast`):

The MD5 sums of all tests match the predefined values for `FLOATTEST8=fp-contract` and `FLOATTEST12=fp-contract`.  This is consistent with the default behavior of GCC when compiling AArch64, PowerPC, and RISC-V code.  (Refer to #710.)

`-ffp-contract=off`:

The MD5 sums of all tests match the predefined values for `FLOATTEST8=no-fp-contract` and `FLOATTEST12=no-fp-contract`.

Conclusion
----------

As a result of this research, I have modified CMakeLists.txt so that, when Clang is used rather than GCC, the default values of `MD5_PPM_3x2_FLOAT_FP_CONTRACT` are changed accordingly.  Thus, setting `FLOATTEST8` and `FLOATTEST12` to `fp-contract` should now work when Clang 14 is used with no arguments, or when Clang 13- is used with `-ffp-contract=on` to build for an architecture that has FMA instructions.

**Comment by dcommander** [comment]
If the default in GCC is changed from `-ffp-contract=fast` to `-ffp-contract=standard`, then I will probably have to revisit the MD5 sums associated with `FLOATTEST8=fp-contract` and `FLOATTEST12=fp-contract`, but for now, everything should at least work with GCC up to 13 and Clang up to 14.
